samples/ts/21.react-media-template/src/components/PlayerProgressBar.tsx

```diff
@@ -0,0 +1,209 @@
+            setDimensions(sliderRef.current.getBoundingClientRect());
+        }
+    }, [sliderRef, duration, width, height]);
+
+    const onMouseMove: ReactEventHandler<HTMLElement> = (
+        e: SyntheticEvent<HTMLElement, MouseEvent>
+    ) => {
+        if (dimension) {
+            const xPosition = Math.min(
+                Math.max(e.nativeEvent.clientX, dimension.left),
+                dimension.right - 4
+            );
+            const distanceFromOrigin = xPosition - dimension.left;
+            const mousePos = (distanceFromOrigin / dimension.width) * 100;
+            const hoverTime = Math.max(
+                0,
+                Math.min(Math.round(duration * (mousePos / 100)), duration)
+            );
+            const scrollOffSet = 0;
+
+            setToolTipContent(formatTimeValue(hoverTime));
+
+            toolTipPositioningRef.current?.setTarget({
+                getBoundingClientRect: getRect(
+                    xPosition,
+                    dimension.top - scrollOffSet
+                ),
+            });
+        }
+    };
+
+    const onTouchMove: ReactEventHandler<HTMLElement> = (
+        e: SyntheticEvent<HTMLElement, TouchEvent>
+    ) => {
+        if (dimension) {
+            const xPosition = Math.min(
+                Math.max(e.nativeEvent.touches[0].clientX, dimension.left),
+                dimension.right - 4
+            );
+            const distanceFromOrigin = xPosition - dimension.left;
+            const mousePos = (distanceFromOrigin / dimension.width) * 100;
+            const hoverTime = Math.max(
+                0,
+                Math.min(Math.round(duration * (mousePos / 100)), duration)
+            );
+            const scrollOffSet = 0;
+
+            setToolTipContent(formatTimeValue(hoverTime));
+
+            toolTipPositioningRef.current?.setTarget({
+                getBoundingClientRect: getRect(
+                    xPosition,
+                    dimension.top - scrollOffSet
+                ),
+            });
+        }
+    };
+
+    const durationToDivideBy = duration === 0 ? 100 : duration;
+    const bufferLoadedPercent = 0;
+
+    return (
+        <div className={flexItemStyles.noShrink} ref={resizeRef}>
+            <div className={styles.pageEl}>
+                <Tooltip
+                    withArrow
+                    positioning={{
+                        positioningRef:
+                            toolTipPositioningRef as MutableRefObject<PositioningImperativeRef>,
+                    }}
+                    content={toolTipContent}
+                    relationship="label"
+                >
+                    <Slider
+                        ref={sliderRef as MutableRefObject<HTMLInputElement>}
+                        min={0}
+                        max={durationToDivideBy}
+                        value={localCurrentTime}
+                        disabled={isPlaybackDisabled}
+                        // TODO: ts errors
```

Comment:
Maybe take a look at the YouTube app to see if it's resolved there? Otherwise add an eslint exception